Add disable-builtin-k8s-secret-store annotation option

DeepanshuA opened the pull request:
Signed-off-by: Deepanshu Agarwal <deepanshu.agarwal1984@gmail.com>

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Dapr currently, by default, enables a built-in kubernetes secret store in kubernetes mode.
But, considering that an application developer may not want to use any Secrets API or may want to use a different Secret Store, there should be an option to disable it as well. That is why a new setting `disable-builtin-k8s-secret-store` is provided, which can be set to `true`, if required. By default, it is kept as `false` to not disturb the on-going default behavior of Dapr.


## Issue reference

Please reference the issue this PR will provide documentation to: Issue: [#dapr/dapr 4227](https://github.com/dapr/dapr/issues/4227) & PR: https://github.com/dapr/dapr/pull/4605

DeepanshuA committed:
Add disable-builtin-k8s-secret-store annotation option

Signed-off-by: Deepanshu Agarwal <deepanshu.agarwal1984@gmail.com>

msfussell commented:
@DeepanshuA - First, thanks for this contribution. Can you link this issue to the code PR/Issue?
With many of these options it is good to explain the "why" to when to use to option, so people have a fuller context. In looking at the secrets overview section here https://docs.dapr.io/developing-applications/building-blocks/secrets/secrets-overview/ I see that the docs really do not described that a K8s secret store is created for you and without understanding this first, it is hard to understand this setting.

My ask is, could you update the secrets overview article above as part of this PR to explain the creation of this K8s secrets strore, when you would and would not use this and then reference this setting in the scenarios where this should not be made available to developers? Would greatly appreciate this and your time.

greenie-msft committed:
Update Preview Features List (#2555)

Updated resiliency to have a version introduced and fixed version introduced for pubsub routing

Signed-off-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>

DeepanshuA committed:
Add k8s secret store description in secrets-overview

Signed-off-by: Deepanshu Agarwal <deepanshu.agarwal1984@gmail.com>

DeepanshuA commented:
Thanks for the guidance @msfussell. It surely makes sense to add more details to be able to convey the significance of these ongoing new annotations being added. 
Please check if the addition with new commit looks fine.

hhunter-ms commented on daprdocs/content/en/developing-applications/building-blocks/secrets/secrets-overview.md:
```diff
@@ -17,6 +17,9 @@ Using Dapr's secret store building block typically involves the following:
 1. Retrieving secrets using the Dapr secrets API in the application code.
 1. Optionally, referencing secrets in Dapr component files.
 
+Note: Dapr currently, by default, enables a built-in kubernetes secret store in kubernetes mode.
```
```suggestion
{{% alert title="Note" color="primary" %}}
By default, Dapr currently enables a built-in Kubernetes secret store in Kubernetes mode. If you don't want to use any secrets API or would rather use a different secret store, you can disable the default Kubernetes secret store using the new `disable-builtin-k8s-secret-store` setting, which can be set to `true`. The default setting is `false`, to not disturb the ongoing default behavior of Dapr.
{{% /alert %}}
```

Encapsulating the whole note in shortcode

msfussell replied:
I would suggest a slight variant like this.

```suggestion
{{% alert title="Note" color="primary" %}}
By default, Dapr currently enables a built-in Kubernetes secret store in Kubernetes mode (deployed via the Helm defaults or `dapr init -k`). If you are using another secret store, you can disable the Dapr Kubernetes secret store using the `disable-builtin-k8s-secret-store` setting, when set to `true`. The default is `false`.
{{% /alert %}}
```

DeepanshuA replied:
Thanks @hhunter-ms  and @msfussell. Incorporated suggestions.

msfussell commented:
@DeepanshuA - Just seen this is a new feature in v1.8 and so needs to target the v1.8 branch. can you update (or create a new PR) to point at that docs branch? Thanks

DeepanshuA committed:
Incorporating review comments from docs#2548

Signed-off-by: Deepanshu Agarwal <deepanshu.agarwal1984@gmail.com>

DeepanshuA committed:
Correcting Docs

Signed-off-by: Deepanshu Agarwal <deepanshu.agarwal1984@gmail.com>

DeepanshuA commented:
> @DeepanshuA - Just seen this is a new feature in v1.8 and so needs to target the v1.8 branch. can you update (or create a new PR) to point at that docs branch? Thanks

Thanks for correction. Moved branch to v1.8.

msfussell committed:
Merge branch 'v1.8' into disable-k8s-store

msfussell commented on daprdocs/content/en/reference/arguments-annotations-overview.md:
```diff
@@ -58,4 +58,5 @@ This table is meant to help users understand the equivalent options for running
-| not supported | not supported | | `dapr.io/volume-mounts-rw` | List of pod volumes to be mounted to the sidecar container in read-write mode. Strings consisting of `volume:path` pairs separated by a comma. Example, `"volume-1:/tmp/mount1,volume-2:/home/root/mount2"`. |
\ No newline at end of file
+| not supported | not supported | | `dapr.io/volume-mounts-rw` | List of pod volumes to be mounted to the sidecar container in read-write mode. Strings consisting of `volume:path` pairs separated by a comma. Example, `"volume-1:/tmp/mount1,volume-2:/home/root/mount2"`. |
+| `--disable-builtin-k8s-secret-store` | not supported | | `dapr.io/disable-builtin-k8s-secret-store` | Disables BuiltIn Kubernetes secret store. Default value is false. |
```
```suggestion
| `--disable-builtin-k8s-secret-store` | not supported | | `dapr.io/disable-builtin-k8s-secret-store` | Disables BuiltIn Kubernetes secret store. Default value is false. See [Kubernetes secret store component]({{<ref "kubernetes-secret-store.md">}}) for details. |
```

msfussell commented on daprdocs/content/en/developing-applications/building-blocks/secrets/secrets-overview.md:
```diff
@@ -17,6 +17,10 @@ Using Dapr's secret store building block typically involves the following:
 1. Optionally, referencing secrets in Dapr component files.
 
+{{% alert title="Note" color="primary" %}}
+By default, Dapr currently enables a built-in Kubernetes secret store in Kubernetes mode (deployed via the Helm defaults or `dapr init -k`). If you are using another secret store, you can disable the Dapr Kubernetes secret store using the `disable-builtin-k8s-secret-store` setting, when set to `true`. The default is `false`.
```
```suggestion
By default, Dapr enables a built-in [Kubernetes secret store in Kubernetes mode]({{<ref "kubernetes-secret-store.md">}}) deployed either via the Helm defaults or `dapr init -k`. If you are using another secret store, you can disable (not configure) the Dapr Kubernetes secret store using the `disable-builtin-k8s-secret-store` setting, when set to `true` through the Helm settings. The default is `false`.
```

msfussell committed:
Update daprdocs/content/en/developing-applications/building-blocks/secrets/secrets-overview.md

Signed-off-by: Mark Fussell <markfussell@gmail.com>

msfussell committed:
Update daprdocs/content/en/reference/arguments-annotations-overview.md

Signed-off-by: Mark Fussell <markfussell@gmail.com>

msfussell reviewed (approved):
LGTM
